### PR TITLE
Handle AI service exceptions when starting second instance

### PR DESCRIPTION
**Pull request by Siedlerchr**
Fixes https://github.com/JabRef/jabref-issue-melting-pot/issues/575

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Review by koppor — CHANGES_REQUESTED** [review]
The code can IMHO be simplified:

Add a try catch in `MVStoreBase` and reinitalize MVStore with `fileNAme´`null`.

Reason:

	at org.jabref@100.0.0/org.jabref.logic.ai.util.MVStoreBase.<init>(MVStoreBase.java:32)

And also the existing line 31.

CC @InAnYan

**Comment by Siedlerchr** [comment]
If I pass null for filename it will result in an IllegalArgumentException – if the directory does not exist  
this is from the internal MVStore constructor 
grafik

**Comment by Siedlerchr** [comment]
Ah I see, null makes it in memory

**Comment by koppor** [comment]
> Ah I see, null makes it in memory

Commemt in code missing maybe 🙈🙈🙈

**Comment by InAnYan** [comment]
@Siedlerchr sorry, this looks a little bit wrong.

What I tried to do in my original code:
- If there are no errors: open file MVStore.
- If there are some errors: tell users about the error (`dialogService` + `LOGGER`) + **open MVStore in RAM** (by passing path to null).

The exception was:
```
Caused by: org.h2.mvstore.MVStoreException: The file is locked: ...
```

It should be handled gracefully the way I described above...

**Comment by InAnYan** [comment]
And I opened a RAM MVStore for 2 reasons:
- The ability to continue using JabRef
- Not using `null`s

**Comment by InAnYan** [comment]
I wonder what will happen in search with Postgres, if there are 2 JabRef instances

**Comment by Siedlerchr** [comment]
changed it now to use null, makes it easier

**Comment by koppor** [comment]
> I wonder what will happen in search with Postgres, if there are 2 JabRef instances

On each change, an update will be pushed to the SQL database. Thus, the search maches the new data. However, the data is not present in the main table in the other instance. Inconsistencies ^^. Not sure whether we should spend effort on that. Current work around would be to add the process id of the current jabref instance as table name, since we build the table on each start. But if we want to persist the index across sessions, this workaround does not work any more. -- Too much effort for such a seldom case, --> future work. cc @LoayGhreeb

File: src/main/java/org/jabref/logic/ai/chatting/chathistory/storages/MVStoreChatHistoryStorage.java [hunk 1]
```diff
@@ -119,10 +119,6 @@ private Map<Integer, ChatHistoryRecord> getMap(Path bibDatabasePath, String type
         return mvStore.openMap(bibDatabasePath + "-" + type + "-" + name);
     }
 
-    public void commit() {
-        mvStore.commit();
-    }
-
```
Comment (koppor) [inline src/main/java/org/jabref/logic/ai/chatting/chathistory/storages/MVStoreChatHistoryStorage.java:125]: I did not see any discussion why a removal of that is OK. @InAnYan ?

Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/ai/chatting/chathistory/storages/MVStoreChatHistoryStorage.java:125]: This is just an overriden method

Comment (koppor) [reply src/main/java/org/jabref/logic/ai/chatting/chathistory/storages/MVStoreChatHistoryStorage.java:125]: Sometimes, this diff-based reviewing is shitty ^^

**Review by koppor — APPROVED** [review]

**Pull request merged**